kernel/work.c

```diff
@@ -671,6 +671,14 @@ static void work_queue_main(void *workq_ptr, void *p2, void *p3)
 		}
 
 		if (work == NULL) {
+			__maybe_unused int backup;
+
+			/* Disable */
+			if (IS_ENABLED(CONFIG_KERNEL_WARN_LONG_TIME_PENDING)) {
+				backup = _current->long_time_warns;
+				_current->long_time_warns = -1;
+			}
```

Comment:
You can't mix these checks with the `#ifdef`s elsewhere, as the `long_time_warns` field won't exist.

Reply:
Thanks.